tests/lean/rwPrioritizesLCtxOverEnv.lean

```diff
@@ -0,0 +1,13 @@
+/-! # Ensure `rw` prioritizes the local context over the environment
+
+This test ensures that `rw [foo]` looks for `foo` in the local context before it looks for a
+constant named `foo` in the environment. See issue #2729.
+-/
+
+-- Introduce `foo` to the environment.
+/-- A constant we will *not* be rewriting with. -/
+def foo : List Nat := [0]
+
+example (A B : Prop) (foo : A ↔ B) (b : B) : A := by
+  rw [foo] -- should be interpreted as `foo : A ↔ B`, not `foo : List Nat`, and succeed
+  assumption
```

Comment:
I think we need a test that shows that we can still rewrite by the global constant, by fully specifying it (e.g. with `_root_`).

Reply:
Good point; added in c0e3492 (rebased for nicer commit history). :)